### ISSUE TITLE
Help to Revert to previous version

### ISSUE THREAD

**Issue by ksdavidc** [JohnCoates/Aerial#1259]
I unfortunately am using an old High Sierra system. I was working fine with (I believe) 3.09 installed, but I accidentally updated via the companion to 3.1 and now it doesn't work. I downloaded/installed the 3.09 and 3.06 versions of aerial, but if I open the companion it gives a message saying it needs the latest version...but it used to work...Can I get it to work again?

**Comment by mvorl** [comment]
In my High Sierra installation, after the update to 3.10 it showed a moving text telling me that "this version of Aerial does not work with your version of MacOS".

I reverted to 3.09 by downloading the kit directly, and after unzipping it, copy it directly to ~/Library/Screen Savers/ .

**Comment by ksdavidc** [comment]
Strange. I was just about to respond this, looked over at my screen, and it was working.

Thank you for the kind replies. 

I did the same thing @mvori did, i.e. download the 3.09 zip and install manually, but was getting the moving text they describe, and the screensaver was inoperative. It seems to have fixed itself, aka showing the screensaver, no message, and companion is telling me 3.1 is ready. 

If I remember correctly at one point running the companion gave me a message about being installed for all users, so it needed to upgrade, but I am not sure if that was originally or during one of my attempts to fix it, which involved various deletes and reinstalls. Eventually I installed it for one user only. Sorry not to be able to give a more detailed breakdown. I should have been less happy-go-lucky.

The companion seems to be working for the most part. I can open the settings, for example. Launch now isn't working, but it may be something else on my computer is wonky.

**Comment by glouel** [comment]
So, a few things @mvorl & @ksdavidc 

First, the "You cannot use the Aerial screen saver with this version of macOS." is not exactly a new bug. It can happen anytime you update Aerial while having your System Preferences panel open, or if you switch between screen savers (on anything <10.15). The quick fix is, go to System Preferences, make sure Aerial is selected, close System Preferences (the whole window) and reopen. I have more details here if you are interested, see the first "Very common issues..." at the bottom of this page : https://aerialscreensaver.github.io/troubleshooting.html

So I would make sure that whichever version you try, you are aware of this. 

Also, if you have multiple version of Aerial showing up in the Screen saver panel, remove them all, and reinstall fresh (don't try to find which one is which, those old versions of macOS will not load them correctly when you have multiple of them). 

It's possible that there's another specific issue with 3.1.0 as it's compiled using the latest Xcode, and Apple periodically breaks/remove support for old versions of macOS. I've confirmed that 3.1.0 "works" with 10.14, but I no longer have anything to test on 10.13 to verify what could be happening. 

If you have issues with auto updates, you can disable them via companion (just say Notify me instead of install automatically in the menu). You can either uninstall companion (trash the app) or just stay like until a fix pops up, if any. 

If you want to manually install, you can get  any version here : https://github.com/JohnCoates/Aerial/releases just grab the zip, close System Preferences, and double click the .saver. When prompted, install for your user only (and not all users).

Finally, and I'm just putting this here, you may want to have a look at something like this patcher that lets you install a version of macOS on unsupported hardware :
http://dosdude1.com/mojave/
http://dosdude1.com/catalina/

Again, use at your own risk, have multiple backups, try installing on a separate drive, etc etc. I do use this to run 10.14 on a 2011 Macbook Air. Extra tip : I wouldn't recommend installing Catalina if you only have 4 GB of RAM. 

If you have more specific info about what fails with 3.1.0 (after making sure it's not the bug at the top), please let me know. I try to do a best effort to support old versions of macOS but Apple makes it really hard, so things will break at some point or another.

**Comment by ksdavidc** [comment]
Wow. That is interesting. 

So I bit the bullet and updated again, with settings closed. It looked like it was working at first. If I loaded full screen, it was ok. (no go on "load now"). When I closed it, though, the companion closed by itself(!). When I opened it again, again I got the scrolling msg, and even full screen did not work. I got a blank screen. I tried opening and closing settings, to no avail. I am trying a restart...[muzak plays. ad for car insurance plays]... nope. no go. will revert.

Have reverted as before, and companion is showing old version, but screen saver is blank and still getting text. Have restarted twice. This is where I was before, when it magically cured itself.

It seems unable to launch aerial settings even from the companion, so I wonder if somehow I have broken the connection to the settings???

I re-downloaded 3.9, and installed by double clicking instead of by moving it manually to the folder. It warned me I was using an older version, and gave me the option to replace, which I did. That seems to have fixed it. It works under 3.9.

I should try that with 3.1 I suppose. So let's do it again. first, via companion. why not, eh.
Okay, it worked the first time I loaded to full screen. I went to settings, and bam, not working again. (again, similar behavior as before)

Okay, I'll try downloading 3.1. I downloaded, unzipped and double clicked. msg: Could not install (nill) screen saver.

I will add that the first time I tried to download/unzip 3.1 it gave me a file corrupted msg. the second time it worked ok. (bug in archiver?) Also I had another screensaver that was giving me the same scrolling text, but i deleted it.

Sticking with 3.9 for now. Maybe next time.

It's a lovely screensaver, my daughter (8) loves all the fish, and it makes us happy to see the weather, the time, and all the lovely images. Thank you, and for being a conscientious developer. :-)

**Comment by mvorl** [comment]
Thanks @glouel. I experienced the same as @ksdavidc and will take the opportunity to bite the bullet to upgrade my trusty old High Sierra installation (MacBookPro8,1 w/ 16GB of RAM) to Mojave. I also **love** the screen saver and it's a worthy reason to upgrade ;-)

**Comment by glouel** [comment]
Very quick update , I did release 3.1.2 yesterday with a very specific settings fix that would have also been an issue on 10.13, but it's likely that this isn't the fix for 10.13 (it only affected saving some settings, it's unlikely this error was the same). I'll try to reinstall 10.13 on my old macbook this weekend and see if I can figure something out. 

> It's a lovely screensaver, my daughter (8) loves all the fish, and it makes us happy to see the weather, the time, and all the lovely images.

Hearing that always make my day ! Give her an extra hug from me next time !

**Comment by rjenness** [comment]
Fwiw I have a MacPro 3,1 running 10.13 thanks to the magic of DosDude's tool. I had a very old version of Aerial running and recently updated to the latest version and was disappointed to find it didn't work any more. Same problems as others have described here. I downloaded 3.09 and I'm up and running again!

Thanks for all your continued work on supporting Aerial. It is my favourite screensaver by far.

**Comment by glouel** [comment]
Just a quick update, I reinstalled High Sierra on an old mac and the issue seems to be that the Swift libraries are not loading under 10.13. Aerial does bundle the swift libraries for older macOS systems to avoid that issue, but they just don't load. I believe Apple may have messed up some compatibility thing in the latest Xcode, or there's a deprecation notice I can't find.

As best as I can tell, the latest version of Xcode has "only" dropped support for 10.12 and still supports deployment for 10.13, but this may not be correct for Swift apps. I tried inspecting the libraries and it looks like they use the 10.14 SDK.

Long story short, I'll try a couple of things and see if I can raise an issue or get clarification from Apple, but this may be the end of the line for 10.13 with the latest Xcode. I'll see if there's something that can be done, or maybe make a build with some older version but at some point that will break too.

**Comment by glouel** [comment]
After a bit more digging, it looks like this may be a plugin (screensaver) specific bug. I'm filling a bug with Apple and will keep all of you posted.

**Comment by rjenness** [comment]
Thanks Guillaume, I really appreciate your efforts to support our poor old High Sierra. 

My Mac Pro 3,1 is supposed to be capped at Mavericks but runs beautifully on High Sierra. I’ve tried to go newer but had problems so I’ve settled for HS and will stick with this OS as long as I can. 

Aerial 3.09 seems to run flawlessly on HS so I’m glad to have a working version even if you’re no longer able to keep it updated for HS. 

Cheers,

Rob

**Comment by webbasan** [comment]
I had the same problem for the same reason (Mac Pro 3,1). 
Downgrading to 3.0.9 helped -- thanks for the tip!

**Comment by glouel** [comment]
So to all, 3.1.3 works again on 10.13, it's complied with Xcode 14 release candidate. However this doesn't mean the issue won't crop up again later. Hopefully Apple fixed the bug that blocked things, but let me know if you hit the issue again in the future.

**Comment by rjenness** [comment]
Thank you, I have updated and all is well again. Appreciate the support!
